peers/app_request_network.go

```diff
@@ -87,7 +91,7 @@ func NewNetwork(
 	metrics, err := newAppRequestNetworkMetrics(registerer)
 	if err != nil {
 		logger.Fatal("Failed to create app request network metrics", zap.Error(err))
-		panic(err)
+		return nil, err
```

Comment:
Should the logger still log a `Fatal` if we're not panicking here anymore?

Reply:
Addressed